File: Data Structures & Algorithms/is-anagram/submission-1.java

class Solution {
    int count(String s,char c){
        int count=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)==c){
                count++;
            }
        }
        return count;
    }
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()){
            return false;
        }
        for(int i=0;i<s.length();i++){
            if(count(s,s.charAt(i))!=count(t,s.charAt(i))){
                return false;
            }
        }
        return true;
        /*
        char[] charsb=s.toCharArray();
        char[] charst=t.toCharArray();
        Arrays.sort(charsb);
        Arrays.sort(charst);
        for(int i=0;i<s.length();i++){
            if(charsb[i]!=charst[i]){
                return false;
            }
        }
        return true;
       */
     
    
    }
}

File: Data Structures & Algorithms/boats-to-save-people/submission-1.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort();count=0
        l=0;r=len(people)-1
        while(l<=r):
            if people[l]+people[r]<=limit:
                count+=1
                l+=1
                r-=1
            elif people[r]<=limit:
                count+=1
                r-=1
            if r==l and people[r]<=limit:
                count+=1
                r-=1
        return count